src/components/TaskItem.vue

```diff
@@ -0,0 +1,68 @@
+<template>
+  <q-item v-ripple clickable data-cy="task_item" @click="$emit('task-clicked', $event, task.t)">
+    <q-checkbox
+      v-model:model-value="task.completed"
+      color="primary"
+      keep-color
+      @update:model-value="$emit('task-completion-toggled', $event, task.t)"
+    />
+
+    <q-item-section>
+      <q-item-label data-cy="task_item_title" lines="2">
+        {{ task.title }}
+      </q-item-label>
+    </q-item-section>
+
+    <q-item-section v-if="task.t.notes" side data-cy="notes_indicator">
+      <q-avatar icon="description">
+        <q-tooltip anchor="center right" self="center left" :offset="[10, 10]">
+          Has additional notes! Click to view.
+        </q-tooltip>
+      </q-avatar>
+    </q-item-section>
+
+    <q-item-section v-if="task.grabPostreqs(incompleteOnly).length" side>
```

Comment:
I'm not sure this is supposed to work. Might need to refactor this as task.hard_postreqs.filter(incompleteOnlyFilter) because it is a cachedTask and I don't think it has direct access to the grabPostreqs function.